tests/transforms/test_op_transform.py

```diff
@@ -0,0 +1,548 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the @op_transform framework"""
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.transforms.op_transforms import OperationTransformError
+
+
+class TestValidation:
+    def test_sphinx_build(self, monkeypatch):
+        """Test that op transforms are not created during Sphinx builds"""
+
+        @qml.op_transform
+        def my_transform(op):
+            return op.name
+
+        assert isinstance(my_transform, qml.op_transform)
+
+        monkeypatch.setenv("SPHINX_BUILD", "1")
+
+        with pytest.warns(UserWarning, match="Operator transformations have been disabled"):
+
+            @qml.op_transform
+            def my_transform(op):
+                return op.name
+
+        assert not isinstance(my_transform, qml.batch_transform)
+
+    def test_error_invalid_callable(self):
+        """Test that an error is raised if the transform
+        is applied to an invalid function"""
+
+        with pytest.raises(
+            OperationTransformError, match="does not appear to be a valid Python function"
+        ):
+            qml.op_transform(5)
+
+    def test_unknown_object(self):
+        """Test that an error is raised if the transform
+        is applied to an unknown object"""
+
+        @qml.op_transform
+        def my_transform(op):
+            return op.name
+
+        with pytest.raises(
+            OperationTransformError,
+            match="Input is not an Operator, tape, QNode, or quantum function",
+        ):
+            my_transform(5)(5)
+
+    def test_empty_qfunc(self):
+        """Test that an error is raised if the qfunc has no quantum operations
+        (e.g., it is not a qfunc)"""
+
+        @qml.op_transform
+        def my_transform(op):
+            return op.name
+
+        def qfunc(x):
+            return x**2
+
+        with pytest.raises(
+            OperationTransformError,
+            match="Quantum function contains no quantum operations",
+        ):
+            my_transform(qfunc)(0.5)
+
+
+class TestUI:

```

Comment:
Class docstring is missing. (this is recurring)

Reply:
```suggestion
class TestUI:
    """Test the user interface of the op_transform, and ensure it applies
    and works well for all combinations of inputs and styles"""
```